diag_manager/fms_diag_object.F90

```diff
@@ -486,62 +488,83 @@ end function fms_diag_axis_init
+  integer                                  :: n1, n2, n3      !< Size of the 3 indicies of the field data
+  integer                                  :: omp_num_threads !< Number of openmp threads
+  integer                                  :: omp_level       !< The openmp active level
+  logical                                  :: buffer_the_data !< True if the user selects to buffer the data and run
+                                                              !! the calculationslater.  \note This is experimental
+  character(len=128)                       :: error_string    !< Store error text
+  logical                                  :: data_buffer_is_allocated !< .true. if the data buffer is allocated
+  character(len=128)                       :: field_info      !< String holding info about the field to append to the
+                                                              !! error message
+  logical, allocatable, dimension(:,:,:,:) :: oor_mask        !< Out of range mask
+  real(kind=r8_kind)                       :: field_weight    !< Weight to use when averaging (it will be converted
+                                                              !! based on the type of field_data when doing the math)
 
 #ifndef use_yaml
 CALL MPP_ERROR(FATAL,"You can not use the modern diag manager without compiling with -Duse_yaml")
 #else
-  class(diagYamlFilesVar_type), pointer :: ptr_diag_field_yaml !< Pointer to a field from yaml fields
+  field_info = " Check send data call for field:"//trim(this%FMS_diag_fields(diag_field_id)%get_varname())
 
-  !TODO: weight is for time averaging where each time level may have a different weight
-  ! call real_copy_set()
+  !< Check if time should be present for this field
+  if (.not.this%FMS_diag_fields(diag_field_id)%is_static() .and. .not.present(time)) &
+    call mpp_error(FATAL, "Time must be present if the field is not static. "//trim(field_info))
 
-  !TODO: oor_mask is only used for checking out of range values.
-  ! call init_mask_3d()
+  !< Set the field_weight. If "weight" is not present it will be set to 1.0_r8_kind
+  field_weight = set_weight(weight)
 
-  !TODO: Check improper combinations of is, ie, js, and je.
-  ! if (check_indices_order()) deallocate(oor_mask)
+  !< Check that the indices are present in the correct combination
+  error_string = check_indices_order(is_in, ie_in, js_in, je_in)
+  if (trim(error_string) .ne. "") call mpp_error(FATAL, trim(error_string)//". "//trim(field_info))
 
-!> Does the user want to push off calculations until send_diag_complete?
+  !< If the field has `mask_variant=.true.`, check that mask OR rmask are present
+  if (this%FMS_diag_fields(diag_field_id)%is_mask_variant()) then
+    if (.not. associated(mask) .and. .not. associated(rmask)) call mpp_error(FATAL, &
+      "The field was registered with mask_variant, but mask or rmask are not present in the send_data call. "//&
```

Comment:
Can we include the field name in these error messages?  That will make debugging easier if an error is encountered.

Reply:
The field name is in `field_info`
field_info = " Check send data call for field:"//trim(this%FMS_diag_fields(diag_field_id)%get_varname())